### PR TITLE
[DomCrawler] No more exception on field name with strange format

### PR DESCRIPTION
**Pull request by guiled**
| Q | A |
| --- | --- |
| Branch? | "master" for new features / 2.3, 2.7, 2.8 or 3.0 for fixes |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes/no |
| Fixed tickets | #18569 |
| License | MIT |
| Doc PR | reference to the documentation PR, if any |

### CODE REVIEW

**Comment by guiled** [comment]
Tests are failing, but this PR can be accepted I will need some help to manage all the tests about malformed name...
Thank you.

File: src/Symfony/Component/DomCrawler/FormFieldRegistry.php [hunk 1]
```diff
@@ -207,16 +207,17 @@ private function getSegments($name)
...
-                    throw new \InvalidArgumentException(sprintf('Malformed field path "%s"', $name));
+                    $segments[] = $extra;
                 }
             }
 
             return $segments;
         }
 
-        throw new \InvalidArgumentException(sprintf('Malformed field path "%s"', $name));
+        return [$name];
```
Comment (fabpot) [inline src/Symfony/Component/DomCrawler/FormFieldRegistry.php]: should be `array($name)`

**Comment by stof** [comment]
this cannot be accepted without tests covering these cases to prevent regressions

**Comment by guiled** [comment]
I think that the things I will do is to remove the tests that check if the DomCrawler raise error when the name is malformed.

**Comment by nicolas-grekas** [comment]
@guiled remove these tests & move the used-to-be invalid to a test ensuring they now are accepted :)

**Comment by fabpot** [comment]
@guiled do you have time to finish this PR?

**Comment by fabpot** [comment]
I'm taking over to finish this one. See #19138

**Pull request closed without merging**